belief_manager.py

```diff
@@ -0,0 +1,578 @@
+import collections
+import logging
+import sys
+
+import numpy as np
+
+from IPC.socket_ipc_factory import SocketIPCFactory
+
+from kalman import Kalman
+import statistics
+
+
+logger = logging.getLogger(__name__)
+
+
+class BeliefManager(object):
+  """ Deals with high-level belief system control, including sending and receiving
+  data as well as interfacing with the Kalman filter. """
+  # Constants for x and y indices in coordinate tuples.
+  _X = 0
+  _Y = 1
+
+  # Z-score to use when calculating error regions.
+  ERROR_REGION_Z_SCORE = 2.576
+  # The minimum distance apart (m) two points need to be before we'll use them
+  # to calculate a transmitter position.
+  MIN_DISTANCE = 15
+  # The number of cycles we have to go without receiving a reading from a
+  # particular transmitter in order to deem that transmitter nonexistent.
+  MAX_INNACTIVE_CYCLES = 10
+  # How far away (m) we can be from a transmitter signal and still register it,
+  # in average conditions.
+  RADIO_RANGE = 40.0
+
+  def __init__(self):
+    self._initialize_member_variables()
+
+    # Initialize connections to the Pixhawk reader and wireless
+    # communications handler, as well as the transmitter locator.
+    factory = SocketIPCFactory()
```

Comment:
Will have to re-factor this later to use Multiprocessing. I suggest taking queues you need as arguments to constructor.